packages/openapi-to-graphql/src/resolver_builder.ts

```diff
@@ -40,13 +49,272 @@ type GetResolverParams = {
   operation: Operation
   argsFromLink?: { [key: string]: string }
   payloadName?: string
+  responseName?: string
   data: PreprocessingData
   baseUrl?: string
   requestOptions?: NodeRequest.OptionsWithUrl
 }
 
+type GetSubscribeParams = {
+  operation: Operation
+  argsFromLink?: { [key: string]: string }
+  payloadName?: string
+  data: PreprocessingData
+  baseUrl?: string
+  connectOptions?: ConnectOptions
+}
+
+/*
+ * If operationType is Subscription, creates and returns a resolver object that contains subscribe to perform subscription
+ * and resolve to execute payload transformation
+ */
+export function getSubscribe({
+  operation,
+  argsFromLink = {},
+  payloadName,
+  data,
+  baseUrl,
+  connectOptions
+}: GetSubscribeParams): SubscriptionIterator {
+  // Determine the appropriate URL:
+  if (typeof baseUrl === 'undefined') {
+    baseUrl = Oas3Tools.getBaseUrl(operation)
+  }
+
+  // Return custom resolver if it is defined
+  const customResolvers = data.options.customResolvers
+  const title = operation.oas.info.title
+  const path = operation.path
+  const method = operation.method
+
+  if (
+    typeof customResolvers === 'object' &&
+    typeof customResolvers[title] === 'object' &&
+    typeof customResolvers[title][path] === 'object' &&
+    typeof customResolvers[title][path][method] === 'object' &&
+    customResolvers[title][path][method].hasOwnProperty('subscribe')
+  ) {
+    translationLog(
+      `Use custom subscribe resolver for ${operation.operationString}`
+    )
+    const customResolver = customResolvers[title][path][method] as ResolveObject
+    if (typeof customResolver.subscribe === 'function') {
+      return customResolver.subscribe
+    }
+  }
+
+  return (root: any, args, ctx, info = {}) => {
+    /**
+     * Determine possible topic(s) by resolving callback path
+     *
+     * GraphQL produces sanitized payload names, so we have to sanitize before
+     * lookup here
+     */
+
+    const paramName = Oas3Tools.sanitize(
+      payloadName,
+      Oas3Tools.CaseStyle.camelCase
+    )
+
+    let resolveData: any = {}
+
+    if (
+      root &&
+      typeof root === 'object' &&
+      typeof root['_openAPIToGraphQL'] === 'object' &&
+      typeof root['_openAPIToGraphQL'].data === 'object'
+    ) {
+      const parentIdentifier = getParentIdentifier(info)
+      if (
+        !(parentIdentifier.length === 0) &&
+        parentIdentifier in root['_openAPIToGraphQL'].data
+      ) {
+        /**
+         * Resolving link params may change the usedParams, but these changes
+         * should not be present in the parent _openAPIToGraphQL, therefore copy
+         * the object
+         */
+        resolveData = JSON.parse(
+          JSON.stringify(root['_openAPIToGraphQL'].data[parentIdentifier])
+        )
+      }
+    }
```

Comment:
Not sure if the above should be kept ?

Reply:
This is only necessary for resolving links... I'm not entirely sure what that means in respect to callbacks.

Reply:
Right, now that i know its purpose i will remove it.